lock-api-objects/src/main/java/com/palantir/lock/watch/TransactionsLockWatchEvents.java

```diff
@@ -50,7 +50,7 @@ static ForcedSnapshot failure(LockWatchStateUpdate.Snapshot snapshot) {
         @Value.Parameter
         List<LockWatchEvent> events();
         @Value.Parameter
-        Map<Long, Long> startTsToSequence();
+        Map<Long, IdentifiedVersion> startTsToSequence();
```

Comment:
I'm just wondering if this map should actually be on both results of this method, and here is why:

The client of Atlas may receive a snapshot because their version is too old, but the version that each transaction holds is definitely not too old (by the definition of our cache system). Therefore, it surely makes sense that we can return a snapshot that resets the cache state, along with all the events that have happened between that snapshot and the latest version in the timestamp map. This would mean that the only difference between success and failure is that failure contains a snapshot also. Since a snapshot is essentially just a `LockWatchCreatedEvent`, this would mean that we could condense this interface to have a single return type, but with a boolean flag of whether the cache needs to be reset or not.

@jkozlowski @j-baker thoughts? This would make more sense with the Alta implementation too (simplifying the types and so forth).

Reply:
Chatted with @jkozlowski offline, and we think it's the right thing to do. Will change that now.